server/src/main/java/org/apache/iotdb/db/utils/datastructure/TVList.java

```diff
@@ -531,6 +530,17 @@ public boolean hasNextTimeValuePair() {
       return hasCachedPair;
     }
 
+    private boolean isPointDeleted(long timestamp) {
+      if (deletionList != null) {
+        for (Pair<Long, Long> del : deletionList) {
+          if (del.left <= timestamp && timestamp <= del.right) {
+            return true;
+          }
+        }
```

Comment:
I think it is not necessary to start from the first of the deletions at each call of `isPointDeleted()`, as this method is called with ordered timestamps, if one timestamp is after the range of one deletion, all remaining timestamps will either.

Reply:
Fixed